### ISSUE TITLE
Agent UI without using NodeJS

### ISSUE THREAD

**Issue by justinvan0603** [tompaana/intermediator-bot-sample#7]
Greeting @tompaana, I'm new at Bot framework and I have a plan to do my bot project like your Scenario 2. But I wonder that Instead use Agent UI built in NodeJS, can I just return an <iframe> to my agent for opening a new chat control to communicate with each customer if they want to talk with the agent not with the Bot?

**Comment by tompaana** [comment]
Hi and thanks for the input! This has, more or less, been on my backlog and I'll try to address this as soon as possible. However, I don't mind receiving pull requests in case this is something you'd like to contribute. Cheers!

**Comment by justinvan0603** [comment]
Hello, thanks for your answer. I've researched more and I've got my idea to implement my plan. One more question when I was testing your Bot project was: What happen if I have many Agents and how can I route the Customer's message to specific Agent in situation that "Agent 1" is busy and "Agent 2, 3 or..." is free to talk? Is that according to the AgentID?

**Comment by tompaana** [comment]
In short: The sample doesn't care how you handle that - it's up to your "agent UI".

The way it behaves now is:

1. Agent UI checks for pending requests (in AgentController.cs) and serves the first request it sees
2. Agent UI then sends a backchannel message to the bot and establishes the engagement (see https://github.com/tompaana/intermediator-bot-sample/blob/master/IntermediatorBotSample/MessageRouting/MessageRouterManager.cs#L440)

**Comment by justinvan0603** [comment]
okay, as you've explained to me, as I understood that which Agent will be received message totally depend on how I implement the Agent UI as the sample you've attached in the post?

**Comment by tompaana** [comment]
That's right. Basically the one (agent, instance, web app, whatever) who sends the backchannel message, will own that conversation.

**Comment by justinvan0603** [comment]
okay, thanks for your explaination. It helps me lots.

**Comment by valnav** [comment]
@justinvan0603  So how did you achieve the agent ui, i have a similar issue. any pointers will be helpful.
